include/irr/builtin/glsl/bxdf/brdf/cos_weighted_sample.glsl

```diff
@@ -13,14 +13,27 @@ irr_glsl_BSDFSample irr_glsl_cos_weighted_cos_gen_sample(in irr_glsl_Anisotropic
 
     irr_glsl_BSDFSample smpl;
     smpl.L = L;
-    smpl.probability = dot(interaction.N,smpl.L)*irr_glsl_RECIPROCAL_PI;
+	smpl.LdotN = dot(interaction.N,smpl.L);
```

Comment:
I would like to draw your attention to the linearity of matrix multiplication what you have is:
`N^T . TangentFrame . vec3(p.x,p.y,z)`

given that TangentFrame=TBN is a column matrix, you can see that you can compute
```glsl
LdotT = p.x;
LdotB = p.y;
LdotN = z;
```

Reply:
TODO, but I forgot where I was going with this

Reply:
oh yeah I remember, what I meant is that you need not take dot products with `smpl.L` because `p.xyz` are already our dot products with the 3 basis vectors of the tangent space.